server/src/test/java/com/coincoinche/engine/ContractTest.java

```diff
@@ -0,0 +1,103 @@
+package com.coincoinche.engine;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.coincoinche.engine.cards.Suit;
+
+import org.junit.Test;
+
+/**
+ * Unit tests for the contract.
+ */
+public class ContractTest {
+
+  private class TestCase extends CommonTestHelper.TestCase {
```

Comment:
Can we use clearer naming here? 
```suggestion
  private class ContractEqualsTestCase extends CommonTestHelper.TestCase {
```

I love this helper :p 
Can we add a test for the isHigherThan method as well in this file? It should be really fast with the helper:

```suggestion
  private class ContractEqualsTestCase extends CommonTestHelper.TestCase {
 (...)
 private class ContractIsHigherThenTestCase extends CommonTestHelper.TestCase {
```

Reply:
I'll just move these classes to the methods directly